app/models/xml_source_processor.rb

```diff
@@ -369,6 +369,7 @@ def update_links_and_xml(xml_string, preview_mode=false, text_type)
         article = Article.new
         article.title = title
         article.collection = collection
+        article.created_by = collection.owner
```

Comment:
I'm curious why this is setting the article creator to the collection owner instead of to `current_user`?